src/Microsoft.Extensions.Configuration.AzureAppConfiguration/JsonConfiguration/JsonConfigurationParser.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using Azure.Data.AppConfiguration;
+using Microsoft.Extensions.Configuration.AzureAppConfiguration.AzureKeyVault;
+using Microsoft.Extensions.Configuration.AzureAppConfiguration.FeatureManagement;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Mime;
+using System.Text.Json;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.JsonConfiguration
+{
+    internal static class JsonConfigurationParser
+    {
+        public static bool IsJsonContentType(string contentType)
+        {
+            if (string.IsNullOrWhiteSpace(contentType))
+            {
+                return false;
+            }
+            string acceptedMainType = "application";
+            string acceptedSubType = "json";
+
+            try
+            {
+                ContentType ct = new ContentType(contentType.Trim().ToLower());
+                var type = ct.MediaType;
+                if (type == FeatureManagementConstants.ContentType ||
+                    type == KeyVaultConstants.ContentType)
+                {
+                    return false;
+                }
+
+                if (type.Contains('/'))
+                {
+                    string mainType = type.Split('/')[0];
+                    string subType = type.Split('/')[1];
+                    if (mainType == acceptedMainType)
+                    {
+                        if (subType.Contains('+'))
+                        {
+                            var subTypes = subType.Split('+');
+                            if (Array.Exists(subTypes, x => x == acceptedSubType))
```

Comment:
>Array.Exists(subTypes, x => x == acceptedSubType) [](start = 32, length = 49)

A common way to write this pattern is:

`subtypes.Any(s => s == acceptedSubType);`

Reply:
Your way here works fine if you like. That other way could save code in the future though.

---
In reply to: [454684293](https://github.com/Azure/AppConfiguration-DotnetProvider/pull/181#discussion_r454684293) [](ancestors = 454684293)

Reply:
As suggested by Abhilash, changed it to "(subTypes.Contains(acceptedSubType))" which also works fine.

---
In reply to: [454684520](https://github.com/Azure/AppConfiguration-DotnetProvider/pull/181#discussion_r454684520) [](ancestors = 454684520,454684293)